contracts/tgrade-valset/src/contract.rs

```diff
@@ -540,7 +546,27 @@ fn end_block(deps: DepsMut, env: Env) -> Result<Response, ContractError> {
     let old_validators = VALIDATORS.load(deps.storage)?;
     VALIDATORS.save(deps.storage, &validators)?;
     // determine the diff to send back to tendermint
-    let (diff, update_members) = calculate_diff(validators, old_validators);
+    let (diff, add, remove) = calculate_diff(validators, old_validators);
+    let update_members = RewardsDistribution::UpdateMembers {
+        add: add.clone(),
+        remove: remove.clone(),
+    };
+
+    // update operators list with info about whether or not they're active validators
+    for op in add {
+        operators().update::<_, StdError>(deps.storage, &Addr::unchecked(op.addr), |op| {
+            let mut op = op.ok_or_else(|| StdError::generic_err("operator doesn't exist"))?;
+            op.active_validator = true;
+            Ok(op)
+        })?;
+    }
+    for op in remove {
+        operators().update::<_, StdError>(deps.storage, &Addr::unchecked(op), |op| {
+            let mut op = op.ok_or_else(|| StdError::generic_err("operator doesn't exist"))?;
+            op.active_validator = false;
+            Ok(op)
+        })?;
+    }
```

Comment:
Looks good. Consider using `diff` directly, as mentioned above.

Reply:
Ah, but you need the `operator` key. Makes sense.

Reply:
The alternative then would be to do this update internally, as part of `calculate_diff`. But I think this is good / better for clarity.